### PR TITLE
fix: strict types support

### LINKED ISSUE #108: Strict types support
With Drupal core moving towards strict types, [more info](https://github.com/localgovdrupal/localgov/issues/685) I've added a test pipeline in preparation https://github.com/localgovdrupal/localgov_project/pull/153 which has highlighted some failures which we should fix.

```
 ------ --------------------------------------------------------------------------- 
  Line   modules/contrib/localgov_news/src/NewsExtraFieldDisplay.php                
 ------ --------------------------------------------------------------------------- 
  167    Parameter #1 $form_object of static method                                 
         Drupal\localgov_news\NewsExtraFieldDisplay::articlePromotedStatus()        
         expects Drupal\node\NodeForm, Drupal\Core\Form\FormInterface given.        
  198    Parameter #2 $article of static method                                     
         Drupal\localgov_news\NewsExtraFieldDisplay::articleSetNewsroomPromote()    
         expects Drupal\node\NodeInterface, Drupal\Core\Entity\EntityInterface      
         given.                                                                     
  201    Parameter #2 $article of static method                                     
         Drupal\localgov_news\NewsExtraFieldDisplay::articleUnsetNewsroomPromote()  
         expects Drupal\node\NodeInterface, Drupal\Core\Entity\EntityInterface      
         given.                                                                     
  222    Call to function in_array() requires parameter #3 to be set.               
  253    Call to function array_search() requires parameter #3 to be set.           
 ------ --------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------- 
  Line   modules/contrib/localgov_news/tests/src/Functional/NewsPageTest.php  
 ------ --------------------------------------------------------------------- 
  185    Call to function in_array() requires parameter #3 to be set.         
  195    Call to function in_array() requires parameter #3 to be set.         
  211    Call to function in_array() requires parameter #3 to be set.         
  213    Call to function in_array() requires parameter #3 to be set.         
  215    Call to function in_array() requires parameter #3 to be set.         
  217    Call to function in_array() requires parameter #3 to be set.         
  232    Call to function in_array() requires parameter #3 to be set.         
  234    Call to function in_array() requires parameter #3 to be set.         
  236    Call to function in_array() requires parameter #3 to be set.         
  238    Call to function in_array() requires parameter #3 to be set.         
 ------ --------------------------------------------------------------------- 
```

### PR DESCRIPTION
**Pull request by millnut**
Fix #108

Fixes strict type issues highlighted in the above issue.

### CODE REVIEW

File: src/NewsExtraFieldDisplay.php [hunk 1]
```diff
@@ -135,6 +135,7 @@ public function formAlter(array &$form, FormStateInterface $form_state, $form_id
       ($form_display->getComponent('localgov_news_newsroom_promote'))
     ) {
 
+      /** @var \Drupal\node\NodeForm $form_object */
```
Comment (ekes) [inline src/NewsExtraFieldDisplay.php]: I'd rather use `assert`ions now that we can, rather than IDE magics.

File: localgov_news.module [hunk 2]
```diff
@@ -86,7 +86,9 @@ function localgov_news_field_widget_complete_form_alter(&$field_widget_complete_
     if (count($options) == 1 && isset($options['_none'])) {
       $create_newsroom_path = 'localgov_newsroom';
       if (\Drupal::moduleHandler()->moduleExists('localgov_microsites_group')) {
+        // @phpstan-ignore-next-line
```
Comment (ekes) [inline localgov_news.module]: Does it make sense to comment why the line is ignored?

**Review by ekes — CHANGES_REQUESTED** [review]

File: localgov_news.module (hunk 2 above)
Comment (millnut) [reply localgov_news.module]: Hi @ekes I've added a comment to both lines

File: src/NewsExtraFieldDisplay.php (hunk 1 above)
Comment (millnut) [reply src/NewsExtraFieldDisplay.php]: Hi @ekes I've refactored this so the correct types are used and added a check to make sure `$form_object` is `NodeForm`

**Review by ekes — APPROVED** [review]

**Pull request merged**